src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationAndGeneratorDriverTranslationAction_Actions.cs

```diff
@@ -119,8 +129,8 @@ public override async Task<Compilation> TransformCompilationAsync(Compilation ol
                     return oldCompilation.RemoveAllSyntaxTrees().AddSyntaxTrees(syntaxTrees);
                 }
 
-                // Because this removes all trees, it'd also remove the generated trees
-                public override bool CanApplyToCompilationWithGeneratedDocuments => false;
+                // Because this removes all trees, it'd also remove the generated trees.
```

Comment:
❔ Is it worth expanding the comment to say it could work with a compilation with generated trees if it knew how to remove all _non_-generated trees for such an input?

Reply:
I'm not sure how much value that would have: this remove/re-add we do in two cases:

1. Changing parse options. We'd have to reparse the generated trees at that point no matter what so there's not much left to save.
2. We're reordering documents.

Even if there was RemoveAllSyntaxTrees that only removed the non-generated ones, you're still going to replace all the trees in the end for 1. For 2, we don't really do that in the IDE and I'm not even sure what the compiler could optimize for that.

The changing parse options we _should_ optimize by not parsing files that don't have directives and then not removing/readding them here, but that's for another time.